rancher/util.go

```diff
@@ -1,5 +1,70 @@
 package rancher
 
+import (
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+)
+
+func InterfaceToString(str interface{}) string {
+	switch str.(type) {
+	case string:
+		return str.(string)
+	case float64:
+		return strconv.FormatFloat(str.(float64), 'f', -1, 64)
+	case int:
+		return strconv.Itoa(str.(int))
+	default:
+		logrus.Errorf("Invalid Metadata Key found %s", str)
+	}
+	return ""
+}
+
+func NestedMapsToMapInterface(data map[string]interface{}) map[string]interface{} {
+	newMapInterface := map[string]interface{}{}
+
+	for k, v := range data {
+		switch v.(type) {
+		case map[interface{}]interface{}:
+			v = mapWalk(v.(map[interface{}]interface{}))
+		case []interface{}:
+			if listHasMaps(v.([]interface{})) {
```

Comment:
what if its a list of a list of maps.  Do you really have to check if it has a map?  Just blindy convert.